### PR TITLE
5619 jsonparse  react data

### PR DESCRIPTION
**Pull request by peterbe**
I haven't done any Lighthouse testing locally to see if this makes a big difference. I know it's sane in principle. 

At first, I tried to do ALL of this in Python but after a lot of searching I couldn't find a library that not only JSON serializes but also one that escapes *everything else* that might appear in the string. At first I tried something like this:
```
templ = "<script>window._react_data = JSON.parse('{}');</script>"
return templ.format(
    json.dumps(data).replace('</', '<\\/').replace("'", "\\'") etc etc etc
)
```
But everything I tried just failed. There was always something weird. Especially when involving Unicode. It was just one problem after another. I tried searching for some decent Python libs that does what the npm package `jsesc` does but I couldn't find anything. And I really trust `jsesc` because ita author is really good at this stuff. 

So basically, what this PR does is that it moves the work of serializing the string over the Node instead of Python.

### CODE REVIEW

**Comment by peterbe** [comment]
>  thinning react_data

https://github.com/mozilla/kuma/compare/mozilla:master...peterbe:make-up-_react_data?expand=1

**Comment by Gregoor** [comment]
> 
> 
> > thinning react_data
> 
> https://github.com/mozilla/kuma/compare/mozilla:master...peterbe:make-up-_react_data?expand=1

way ahead of me 🙈

**Comment by peterbe** [comment]
The other option is to consider that hacky trick of using `dangeroulsyInsertSetHTML` with an empty string. But I don't know what that means to hydrating and hooking up to event handlers. 

I know that if you run my branch there, you'll see that it totally solves https://github.com/mozilla/kuma/issues/5613
